tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas1_copy_kk.hpp

```diff
@@ -0,0 +1,59 @@
+
+#ifndef LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_COPY_HPP_
+#define LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_COPY_HPP_
+
+#include "signal_kokkos_impl_called.hpp"
+
+namespace KokkosKernelsSTD {
+
+template<class ExeSpace,
+	 class ElementType_x,
+         std::experimental::extents<>::size_type ... ext_x,
+         class Layout_x,
+         class ElementType_y,
+         std::experimental::extents<>::size_type ... ext_y,
+         class Layout_y>
+void copy(kokkos_exec<ExeSpace> /*kexe*/,
+	  std::experimental::mdspan<
+	    ElementType_x,
+	    std::experimental::extents<ext_x ...>,
+	    Layout_x,
+	    std::experimental::default_accessor<ElementType_x>
+	  > x,
+	  std::experimental::mdspan<
+	    ElementType_y,
+	    std::experimental::extents<ext_y ...>,
+	    Layout_y,
+	    std::experimental::default_accessor<ElementType_y>
+	  > y)
+requires ((sizeof...(ext_x) == sizeof...(ext_y)) && (x.rank() <=2))
+{
+  Impl::signal_kokkos_impl_called("copy");
+
+  auto x_view = Impl::mdspan_to_view(x);
+  auto y_view = Impl::mdspan_to_view(y);
```

Comment:
You have the `Kokkos::View` and an execution space, so would you consider just using `Kokkos::deep_copy` here?

Reply:
I think that if the layout of x and y are different, the deep copy fails. To use deep copy I would need to specialize for when the layouts are the same. Right?

Reply:
done!